src/Security/CheckGroups/CheckGroupServerConfig.js

```diff
@@ -72,6 +72,16 @@ class CheckGroupServerConfig extends CheckGroup {
           }
         },
       }),
+      new Check({
+        title: 'Anonymous Users disabled',
```

Comment:
Curious about your thought process for adding a security warning.

I'm thinking that if someone has an application in which anonymous users are part of their use case, they will always get this warning, even though they may have restricted the permissions of anonymous users accordingly. After all, [anonymous users](https://docs.parseplatform.org/ios/guide/#anonymous-users) is a regular feature of Parse Server:

> Being able to associate data and objects with individual users is highly valuable, but sometimes you want to be able to do this without forcing a user to specify a username and password.
>
> An anonymous user is a user that can be created without a username and password but still has all of the same capabilities as any other PFUser. After logging out, an anonymous user is abandoned, and its data is no longer accessible.

I understand that we disable them by default, as enabling them requires taking security implications into account, but I'm not sure it should be a permanent warning. What do you think?

Reply:
Reading this again, maybe it actually makes sense to add that warning until we change the default in 2024, that's still a long time away. Once the deprecation ended, we could remove the warning.

To my previous comment, I think one should always be able to get down to 0 warnings by improving the configuration. Since anonymous user is a regular feature, any app that uses this would never be able to get to 0 warnings, always indicating that there was a potential security issue, which there is not if classes are properly secured. Or the warning could have additional conditions, e.g. anonymous users enabled + file upload enabled for anonymous user = real vulnerability that warrants a warning.

Reply:
@dblythy what do you think?